### PR TITLE
feat: add support for round operation that mappes to math.roundeven.

### PR DESCRIPTION
**Pull request by ttjost**
It also reorders the operations in `.td` and the lowering pass so that it doesn't cause too many conflicts for followed operations.

### CODE REVIEW

**Review by roberteg16 — APPROVED** [review]

File: test/Conversion/XTenNNToLinalg/Round.mlir [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// RUN: aten-opt --convert-xtennn-to-linalg %s | FileCheck %s
+
+func.func @round(%arg0: tensor<1x10xf32>) -> tensor<1x10xf32> {
+    %0 = xten_nn.round %arg0 : (tensor<1x10xf32>) -> tensor<1x10xf32>
+    return %0 : tensor<1x10xf32>
+// CHECK-LABEL:   func.func @round(
+// CHECK-SAME:                    %[[VAL_0:.*]]: tensor<1x10xf32>) -> tensor<1x10xf32> {
+// CHECK:           %[[VAL_1:.*]] = tensor.empty() : tensor<1x10xf32>
+// CHECK:           %[[VAL_2:.*]] = linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel", "parallel"]} ins(%[[VAL_0]] : tensor<1x10xf32>) outs(%[[VAL_1]] : tensor<1x10xf32>) {
```
Comment (cmcgirr-amd) [inline test/Conversion/XTenNNToLinalg/Round.mlir:10]: what are the affine indexing maps? it should be a 1:1 mapping right?

File: include/xten/Dialect/XTenNN/IR/XTenNNOps.td [hunk 2]
```diff
@@ -219,16 +221,30 @@ def XtenNN_MishOp: XTenNN_Op<"mish", [Pure, TosaExtension, ElementwiseUnary, Sam
...
-def XtenNN_EluOp: XTenNN_Op<"elu", [Pure, TosaExtension, ElementwiseUnary, SameOperandsAndResultElementType]> {
-  let summary = "Calculate the elu operation of the given input tensor element-wise";
+def XtenNN_RoundOp: XTenNN_Op<"round", [Pure, TosaExtension, ElementwiseUnary, SameOperandsAndResultElementType]> {
+  let summary = "Calculate the round operation of the given input tensor element-wise.";
   let description = [{
-    Calculate the elu operation (Exponential Linear Unit) of the given input tensor element-wise.
-    ELU(x) =  x,                  if x >  0
-              alpha * (exp(x)-1), if x <= 0
+    Calculate the round operation of the given input tensor element-wise.
+    In case of halves, the rule is to round them to the nearest even integer as does the 'round' operation in torch and onnx.
```
Comment (cmcgirr-amd) [inline include/xten/Dialect/XTenNN/IR/XTenNNOps.td:228]: ONNX specifies the edge cases for NaN, Infitity and +0/-0 but torch does not afaics. Which do we do here?

File: test/Conversion/XTenNNToLinalg/Round.mlir [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// RUN: aten-opt --convert-xtennn-to-linalg %s | FileCheck %s
+
+func.func @round(%arg0: tensor<1x10xf32>) -> tensor<1x10xf32> {
```
Comment (cmcgirr-amd) [inline test/Conversion/XTenNNToLinalg/Round.mlir:3]: a bf16 cases just for completeness would be great

**Review by cmcgirr-amd — CHANGES_REQUESTED** [review]
bf16 test case just so we can see if it is a supported lowering.

File: test/Conversion/XTenNNToLinalg/Round.mlir (hunk 1 above)
Comment (ttjost) [reply test/Conversion/XTenNNToLinalg/Round.mlir:10]: yes, I am adding them to the check

File: include/xten/Dialect/XTenNN/IR/XTenNNOps.td (hunk 2 above)
Comment (ttjost) [reply include/xten/Dialect/XTenNN/IR/XTenNNOps.td:228]: I think we can assume it is a one-to-one translation from Torch to ONNX (didn't test it but it makes sense). In that case, Torch would be similar to ONNX, and I am thinking that LLVM::RoundEven is dependent on how the target implements.

**Review by cmcgirr-amd — APPROVED** [review]
